### ISSUE TITLE
Tonesque: error when trying to get colors from an image returning a 404

### ISSUE THREAD

**Issue by jeherve** [Automattic/jetpack#1672]
If you try to use Tonesque with an image returning a 404 error, it outputs the following warnings:

```
<b>Warning</b>:  imagecreatefromjpeg(): php_network_getaddresses: getaddrinfo failed: Name or service not known in <b>/wp-content/plugins/jetpack/_inc/lib/tonesque.php</b> on line <b>45</b><br />
<br />
<b>Warning</b>:  imagecreatefromjpeg(http://www.hervedesign.com/wp-content/uploads/2008/10/2365824451_192c8f8088_o.jpg): failed to open stream: php_network_getaddresses: getaddrinfo failed: Name or service not known in <b>/wp-content/plugins/jetpack/_inc/lib/tonesque.php</b> on line <b>45</b><br />
```

Would it be possible to fail silently instead?

**Comment by georgestephanis** [comment]
Is this an internal call that references a url, or a third party theme/plugin using tonesque?

**Comment by jeherve** [comment]
It's a third-party plugin, [color-posts](https://github.com/jeherve/color-posts). Do you think I should handle the 404 in my plugin (probably somewhere around [here](https://github.com/jeherve/color-posts/blob/master/functions.jeherve-get-color.php#L30)?

I do see some room for improvement in our code as well, though. We could check for 404s in `Jetpack_PostImages::get_image`, or in Tonesque?

**Comment by georgestephanis** [comment]
The plugin could just prepend an @ to the calls to avoid 404 errors.  I'd be fine adding some error handling in to core in theory.

**Comment by jeherve** [comment]
Worth noting that this will happen for any errors. Here is the same problem with a 403:
https://twitter.com/helenhousandi/status/768135636230471680

**Comment by jeherve** [comment]
Related discussion:
https://github.com/Automattic/jetpack/pull/3742#discussion_r61244909

**Comment by jeherve** [comment]
This should have been fixed with #34164. Closing.
